src/index.ts

```diff
@@ -30,6 +41,58 @@ if (program.endpointType === "client" && program.host === undefined) {
 
 const endpointType = program.endpointType;
 
+// If we are provided a config file, try reading it
+let hardwareConfig: IPinConfiguration[] | undefined = undefined;
```

Comment:
Maybe a comment here that undefined sets the hardware configuration to default inside the romi-robot would help explain how this works. That or just pass the JSON as part of the constructor so that the logic for hardware is all in romi-robot.